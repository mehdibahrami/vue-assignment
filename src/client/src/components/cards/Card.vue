<template>
  <!-- Card -->
  <div :class="cardClasses">
    <!-- Card header -->
    <div v-if="title" class="card-header">
      <h3>{{ title }}</h3>
    </div>
    <!-- Card body -->
    <div class="card-body">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: "Card",
  props: {
    title: {
      type: String,
    },
    hasPadding: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    cardClasses: function () {
      let classes = "card";

      // Set card classes based on hasPadding prop
      if (this.hasPadding === true) {
        classes += ` card-padding`;
      }

      return classes.trim();
    },
  },
};
</script>

<style scoped lang="scss">
.card {
  width: 100%;
  background-color: $white;
  border-radius: 0.5rem;
  overflow: hidden;
  box-shadow: 0 0 #0000, 0 0 #0000, 0 1px 3px 0 rgba(0, 0, 0, 0.1),
    0 1px 2px 0 rgba(0, 0, 0, 0.06);
}

.card-padding {
  padding: 1rem;
}

.card-header {
  font-size: $font-size-lg;
  line-height: 1.5rem;
  font-weight: $font-weight-medium;
  margin-bottom: 0.75rem;
  color: $gray-dark;
}

.card-body {
  height: 100%;
}
</style>
