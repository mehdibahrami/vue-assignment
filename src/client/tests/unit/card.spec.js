import { shallowMount } from "@vue/test-utils";
import Card from "@/components/cards/Card.vue";

describe("Card", () => {
  let wrapper;

  it("renders correctly", () => {
    // shallow mounting the main component
    wrapper = shallowMount(Card, {
      props: {
        title: "Speed",
      },
    });

    expect(wrapper.find("h3").text()).toBe("Speed");
    expect(wrapper.element).toMatchSnapshot();
  });
});
