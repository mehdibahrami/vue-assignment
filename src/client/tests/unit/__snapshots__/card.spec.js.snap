// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Card -->
  <div
    class="card card-padding"
  >
    <!-- Card header -->
    <div
      class="card-header"
    >
      <h3>
        Speed
      </h3>
    </div>
    <!-- Card body -->
    <div
      class="card-body"
    >
      
      
    </div>
  </div>
  
</div>
`;
